src/main/java/uk/gov/hmcts/sscs/builder/TrackYourAppealJsonBuilder.java

```diff
@@ -37,17 +36,23 @@
     public ObjectNode build(CaseData caseData,
                             RegionalProcessingCenter regionalProcessingCenter) {
 
+        // Create appealReceived eventType for appealCreated CCD event
         List<Event> eventList = caseData.getEvents();
         if (eventList == null || eventList.isEmpty()) {
-            String message = "No events exists for this appeal";
-            CcdException ccdException = new CcdException(new Exception(message));
-            LOG.error(message, ccdException);
-            throw ccdException;
+            if (caseData.getCaseCreated() != null) {
+                caseData = createAppealReceivedEventTypeForAppealCreatedEvent(caseData);
```

Comment:
All the events should be created in CCD. Don't know why it created here. @iguannago is this valid?

Reply:
All the events in TCA are generated based on the eventType.
So here frontend don't understand appealCreated. Thats why we are generating appealReceived eventType for appealCreated. So that frontend display it as appealReceived.